Xamarin.Forms.Core/INavigationPageController.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms
+{
+	public interface INavigationPageController
+	{
+		Stack<Page> StackCopy { get; }
+
+		int StackDepth { get; }
```

Comment:
It is the same, but much faster. `StackCopy` actually constructs a whole new `Stack<Page>` each time it's called.